app/models/mention.rb

```diff
@@ -12,44 +12,21 @@ class Mention < ApplicationRecord
 
   class << self
     def create_all(notifiable)
-      # Only works for comments right now.
-      # Paired with the process that creates the "comment-mentioned-user"
-      @notifiable = notifiable
-      doc = Nokogiri::HTML(notifiable.processed_html)
-      usernames = []
-      mentions = []
-      doc.css(".comment-mentioned-user").each do |link|
-        username = link.text.delete("@").downcase
-        if (user = User.find_by(username: username))
-          usernames << username
-          mentions << create_mention(user)
-        end
-      end
-      delete_removed_mentions(usernames)
-      mentions
+      Mentions::CreateAllJob.perform_later(notifiable.id, notifiable.class.name)
     end
-    handle_asynchronously :create_all
 
-    private
-
-    def delete_removed_mentions(usernames)
-      user_ids = User.where(username: usernames).pluck(:id)
-      mentions = @notifiable.mentions.where.not(user_id: user_ids).destroy_all
-      Notification.remove_each(mentions) if mentions.present?
-    end
-
-    def create_mention(user)
-      mention = Mention.create(user_id: user.id, mentionable_id: @notifiable.id, mentionable_type: @notifiable.class.name)
-      # mentionable_type = model that created the mention, user = user to be mentioned
-      Notification.send_mention_notification(mention)
-      mention
+    def create_all_without_delay(notifiable)
+      Mentions::CreateAllJob.perform_now(notifiable.id, notifiable.class.name)
     end
   end
 
   def send_email_notification
-    NotifyMailer.new_mention_email(self).deliver if User.find(user_id).email.present? && User.find(user_id).email_mention_notifications
+    Mentions::SendEmailNotificationJob.perform_later(id) if User.find(user_id).email.present? && User.find(user_id).email_mention_notifications
```

Comment:
I suggest caching `User.find(user_id)` in a local variable. This way less sql queries would be made and it would also be easier to understand.